library/std/src/io/error.rs

```diff
@@ -156,6 +191,62 @@ pub enum ErrorKind {
     /// [`Ok(0)`]: Ok
     #[stable(feature = "rust1", since = "1.0.0")]
     WriteZero,
+    /// The underlying storage (typically, a filesystem) is full.
+    ///
+    /// This does not include out of quota errors.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    NoStorageSpace,
+    /// Invalid seek
+    ///
+    /// Seeking was attempted on an open file handle which is not suitable for seeking - for
+    /// example, on Unix, a named pipe opened with `File::new`.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    InvalidSeek,
+    /// Filesystem quota was exceeded.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    FilesystemQuotaExceeded,
+    /// File larger than allowed or supported.
+    ///
+    /// This might arise from a hard limit of the underlying filesystem or file access API, or from
+    /// an administratively imposed resource limitation.  Simple disk full, and out of quota, have
+    /// their own errors.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    FileTooLarge,
+    /// Resource is busy.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    ResourceBusy,
+    /// Executable file is busy.
+    ///
+    /// An attempt was made to write to a file which is also in use as a running program.  (Not all
+    /// operating systems detect this situation.)
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    ExecutableFileBusy,
```

Comment:
See ResourceBusy

Reply:
On Unix we want to distinguish EBUSY which typically means the system is short of resources or something, from ETXTBSY which means the executable is running.  I'm not sure how to map that to Windows.

Reply:
Yeah, I point to ResourceBusy for my comment, Windows has a whole host of Busy errors, but I'm not sure if any map the same as Unix does.

Reply:
For now I will map `ERROR_BUSY` to `ResourceBusy`.  I'm not sure whether to fold `ETXTBUSY` and `ERROR_PATH_BUSY` together.  In general Unix doesn't enforce any kind of mutual exclusion on files, but Windows does.  `ETXTBSY` is one of Unix's few exceptions (and doesn't appear on every Unix, either).